Caltech-IPAC/firefly
# Firefly 37 recognizing s_region in a table colum

https://jira.ipac.caltech.edu/browse/FIREFLY-42 (this PR is for FIREFLY-42, not 37)
https://jira.lsstcorp.org/browse/DM-18510
this PR includes development on

* create two drawing layers for the table has columns compliant to ObsCore table with s_region column:
    * observation center layer contain point objects for s_ra & s_dec column, like normal catalog layer
    * observation region layer contain region objects for s_region column
* the layer dialog shows column info for both observation center and region layers.
* the layer dialog shows buttons on 'all' regions, 'selected' regions and 'highlighted' region for the 
observation region layer
* the layer dialog shows an error icon if any of the region for ‘all’, ’selected’ or ‘highlighted’ selection has invalid or unsupported values in the description.  
* the point or the region object on either layer can be clicked to highlighted, and the the table highlight will appear accordingly. 
* the point on observation center layer can be either selected or filtered using ‘select area’ tool. 
* the region on observation region layer can not be filtered or selected by using ‘select area’ tool. 
* the selection or filtering made on observation center layer will reflect to the region layer and 
the corresponding rows of the table. 
* the selection or filters made on the table will be visible on both observation center and region layers. 

test 
k8s build 
https://irsawebdev9.ipac.caltech.edu/firefly-37/firefly/firefly-dev.html
* do tap search & cone search (m32, 5 arcsec) on the following cases, 
CADC => ivoa => ivoa.Obscore, (all rows with valid s_region value) or  
MAST => ivoa => ivoa.Obscore (some rows with unsupported s_region value)
the coverage view will contain two layers (one for region and one for center) for the table with s_region column. 
* try filter & select on either layer by using 'select area' tool 
* try highlight  by clicking the point or the border of the circle or polygon
* open layer dialog, try 
    * turn on/off the center or the region layer, 
    * for region layer, try the button on different region selection, and check the error message if there is.
